### PR TITLE
appengine/go111/cloudsql: initial commit of updated sample

### PR DESCRIPTION
**Pull request by grayside**

### CODE REVIEW

File: appengine/go11x/cloudsql/app.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+runtime: go111
+
+env_variables:
+  # Replace INSTANCE_CONNECTION_NAME with the value obtained when configuring your
+  # Cloud SQL instance, available from the Google Cloud Console or from the Cloud SDK.
+  # For Cloud SQL 2nd generation instances, this should be in the form of "project:region:instance".
+  #CLOUDSQL_CONNECTION_NAME: INSTANCE_CONNECTION_NAME
+  CLOUDSQL_CONNECTION_NAME: adamross-svls-kibble:us-central1:test
```
Comment (tbpg) [inline appengine/go11x/cloudsql/app.yaml]: Revert.

File: appengine/go11x/cloudsql/cloudsql.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+
+	log.Printf("Listening on port %s", port)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", port), nil))
+}
+
+// GetDB gets a connection to the database.
+func GetDB() *sql.DB {
```
Comment (tbpg) [inline appengine/go11x/cloudsql/cloudsql.go]: Drop the `Get` prefix? https://golang.org/doc/effective_go.html#Getters

File: appengine/go11x/cloudsql/cloudsql.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	var (
+		connectionName = mustGetenv("CLOUDSQL_CONNECTION_NAME")
+		user           = mustGetenv("CLOUDSQL_USER")
+		password       = os.Getenv("CLOUDSQL_PASSWORD") // NOTE: password may be empty
+	)
+
+	// MySQL Connection, comment out for PostgreSQL
```
Comment (tbpg) [inline appengine/go11x/cloudsql/cloudsql.go]: s/for/to use/

Also, period at the end.

File: appengine/go11x/cloudsql/cloudsql.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		http.Error(w, fmt.Sprintf("Could not query db: %v", err), 500)
+		return
+	}
+	defer rows.Close()
+
+	buf := bytes.NewBufferString("Databases:\n")
+	for rows.Next() {
+		var dbName string
+		if err := rows.Scan(&dbName); err != nil {
+			http.Error(w, fmt.Sprintf("Could not scan result: %v", err), 500)
```
Comment (tbpg) [inline appengine/go11x/cloudsql/cloudsql.go]: Same error comment.

File: appengine/go11x/cloudsql/cloudsql.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	if r.URL.Path != "/" {
+		http.NotFound(w, r)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/plain")
+
+	rows, err := db.Query("SHOW DATABASES")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Could not query db: %v", err), 500)
```
Comment (tbpg) [inline appengine/go11x/cloudsql/cloudsql.go]: Maybe log this error and give a user friendly message for the http.Error?

File: appengine/go11x/cloudsql/cloudsql.go [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+
+	log.Printf("Listening on port %s", port)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", port), nil))
+}
+
+// GetDB gets a connection to the database.
```
Comment (tbpg) [inline appengine/go11x/cloudsql/cloudsql.go]: Mention this can panic under [certain conditions].

File: appengine/go11x/cloudsql/cloudsql.go (hunk 2 above)
Comment (grayside) [reply appengine/go11x/cloudsql/cloudsql.go]: ✅

File: appengine/go11x/cloudsql/app.yaml (hunk 1 above)
Comment (grayside) [reply appengine/go11x/cloudsql/app.yaml]: ✅

File: appengine/go11x/cloudsql/cloudsql.go (hunk 3 above)
Comment (grayside) [reply appengine/go11x/cloudsql/cloudsql.go]: ✅

File: appengine/go11x/cloudsql/cloudsql.go (hunk 4 above)
Comment (grayside) [reply appengine/go11x/cloudsql/cloudsql.go]: Rotated these to:

```go
log.Printf("Could not scan result: %v", err)
http.Error(w, "Internal Error", 500)
```

Of note this sample will be replaced by @kurtisvg as part of a broader overhaul.

File: appengine/go11x/cloudsql/app.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+runtime: go111
+
+env_variables:
+  # Replace INSTANCE_CONNECTION_NAME with the value obtained when configuring your
+  # Cloud SQL instance, available from the Google Cloud Console or from the Cloud SDK.
+  # For Cloud SQL 2nd generation instances, this should be in the form of "project:region:instance".
+  CLOUDSQL_CONNECTION_NAME: INSTANCE_CONNECTION_NAME
+  # Replace username and password if you aren't using the root user.
+  CLOUDSQL_USER: root
+  CLOUDSQL_PASSWORD: ''
```
Comment (kurtisvg) [inline appengine/go11x/cloudsql/app.yaml]: Can we put an example password here? to clarify if it's expecting `password` vs `'password'`

Comment (grayside) [reply appengine/go11x/cloudsql/app.yaml]: YAML don't care. Adding a password.

**Pull request merged**